implementations/rust/ockam/ockam/src/profile.rs

```diff
@@ -1,22 +1,28 @@
-use crate::{Contact, OckamError};
+use crate::OckamError;
 use hashbrown::HashMap;
 use ockam_vault_core::{Hasher, KeyIdVault, PublicKey, Secret, SecretVault, Signer, Verifier};
 use std::ops::{Deref, DerefMut};
 use std::sync::{Arc, Mutex};
 
+mod authentication;
+mod contact;
+pub use contact::*;
 mod identifiers;
 pub use identifiers::*;
 mod key_attributes;
 pub use key_attributes::*;
 mod change;
-use crate::history::ProfileChangeHistory;
+use authentication::Authentication;
 pub use change::*;
+use history::ProfileChangeHistory;
 
 pub trait ProfileVault: SecretVault + KeyIdVault + Hasher + Signer + Verifier {}
 
 impl<D> ProfileVault for D where D: SecretVault + KeyIdVault + Hasher + Signer + Verifier {}
 
 pub type ProfileEventAttributes = HashMap<String, String>;
+/// Contacts Database
+pub type ContactsDb = HashMap<ProfileIdentifier, Contact>;
```

Comment:
Should we consider a struct instead of type if we want to handle persistence.?
Doesn't need to be addressed in this PR, just food for thought.

Reply:
I'm sure persistence is something we'll be working on in the future. For now I hope that's fine